src/marlin_pc/mod.rs

```diff
@@ -363,34 +534,36 @@ impl<E: PairingEngine> PolynomialCommitment<E::Fr> for MarlinKZG10<E> {
             )?;
 
             // compute challenge^j and challenge^{j+1}.
-            let challenge_j = opening_challenge.pow([2 * j as u64]);
+            let challenge_j = opening_challenges(opening_challenge_counter);
+            opening_challenge_counter += 1;
 
             assert_eq!(degree_bound.is_some(), rand.shifted_rand.is_some());
 
-            p += (challenge_j, polynomial.polynomial());
-            r += (challenge_j, &rand.rand);
+            p += (challenge_j.clone(), polynomial.polynomial());
+            r += (challenge_j.clone(), &rand.rand);
```

Comment:
You should be able to replace these `clone()`s with dereferences:
```suggestion
            p += (*challenge_j, polynomial.polynomial());
            r += (*challenge_j, &rand.rand);
```

Reply:
the compiler complains for me...

Reply:
Omit the `.clone()` seems to work.